src/main/java/com/google/sps/servlets/AuthenticateServlet.java

```diff
@@ -37,61 +37,33 @@
 import com.google.appengine.api.datastore.Key;
 import com.google.appengine.api.datastore.KeyFactory;
 
-@WebServlet("/add-user")
-public class AddUserServlet extends HttpServlet {
+@WebServlet("/authenticate")
+public class AuthenticateServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse  response) throws IOException {
     response.setContentType("application/json;");
     Gson gson = new Gson();
 
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     UserService userService = UserServiceFactory.getUserService();
-    boolean isUserLoggedIn = userService.isUserLoggedIn();
-
-    if (isUserLoggedIn) {
-      String userEmail = userService.getCurrentUser().getEmail();
-      String userId = userService.getCurrentUser().getUserId();
-      GivrUser currentUser = null;
-
-      boolean doesUserExist = false;
-      boolean isUserMaintainer = false;
-      boolean isLoggedIn = true;
-
-      Filter queryFilter = new FilterPredicate("userId", FilterOperator.EQUAL, userId);
-      Query query = new Query("User").setFilter(queryFilter);
-      
-      FetchOptions fetchOptions = FetchOptions.Builder.withLimit(1);
-      PreparedQuery preparedQuery = datastore.prepare(query);
-      QueryResultList<Entity> userResult = preparedQuery.asQueryResultList(fetchOptions);
-
-      if (userResult.size() < 1) {
-        GivrUser newUser = new GivrUser(userId, isUserMaintainer);
-        currentUser = newUser;
-      
-        Entity userEntity = new Entity("User");
-        userEntity.setProperty("userId", userId);
-        userEntity.setProperty("isMaintainer", isUserMaintainer);
-        datastore.put(userEntity);
-      } else {
-        for (Entity entity: preparedQuery.asIterable(fetchOptions)) {
-          isUserMaintainer = (boolean) entity.getProperty("isMaintainer");
-        }
-        currentUser = new GivrUser(userId, isUserMaintainer);
-      }
+    GivrUser user = GivrUser.getLoggedInUser();
+    boolean isLoggedIn;
+    String url;
+    // User is null when they are not logged in.
+    if (user == null) {
```

Comment:
Instead of null here, could you simplify by merging GiverUser with UserLoginInfo?

Reply:
Would that mean removing the isLoggedIn boolean variable in UserLoginInfo.java, adding the url string variable to GivrUser.java, and removing UserLoginInfo.java completely?

Reply:
What if you moved both the isloggedin and loginurl to giveruser? Then you'd also be able to represent a non-logged in user without using null